feature-libs/checkout/base/core/facade/checkout-delivery-modes.service.spec.ts

```diff
@@ -0,0 +1,325 @@
+    it(`should call the checkoutDeliveryModesConnector.getSupportedModes()`, (done) => {
+      spyOn(connector, 'getSupportedModes').and.callThrough();
+
+      service
+        .getSupportedDeliveryModesState()
+        .pipe(take(1))
+        .subscribe((state) => {
+          expect(state).toEqual({
+            loading: false,
+            error: false,
+            data: mockSupportedDeliveryModes,
+          });
+          expect(connector.getSupportedModes).toHaveBeenCalledWith(
+            mockUserId,
+            mockCartId
+          );
+          done();
+        });
+    });
+
+    it(`should unsuccessfully backOff on Jalo error and put the error in the state`, fakeAsync(() => {
+      spyOn(connector, 'getSupportedModes').and.returnValue(
+        throwError(mockJaloError)
+      );
+
+      let resultState: QueryState<DeliveryMode[] | undefined> | undefined;
+      const subscription = service
+        .getSupportedDeliveryModesState()
+        .subscribe((result) => (resultState = result));
+
+      tick(4200);
+
+      expect(resultState).toEqual({
+        loading: false,
+        error: <Error>mockJaloError,
+        data: undefined,
+      });
+      subscription.unsubscribe();
+    }));
+
+    it(`should successfully backOff on Jalo error and recover after the 2nd attempt`, fakeAsync(() => {
+      spyOn(connector, 'getSupportedModes').and.returnValues(
+        // first attempt
+        throwError(mockJaloError),
+        // second attempt
+        throwError(mockJaloError),
+        // third time the charm
+        of(mockSupportedDeliveryModes)
+      );
+
+      let resultState: QueryState<DeliveryMode[] | undefined> | undefined;
+      const subscription = service
+        .getSupportedDeliveryModesState()
+        .subscribe((result) => (resultState = result));
+
+      // 1*1*300 = 300
+      tick(300);
+      expect(resultState).toEqual({
+        loading: true,
+        error: false,
+        data: undefined,
+      });
+
+      // 2*2*300 = 1200
+      tick(1200);
+      expect(resultState).toEqual({
+        loading: false,
+        error: false,
+        data: mockSupportedDeliveryModes,
+      });
+      subscription.unsubscribe();
+    }));
+  });
+
+  describe(`getSelectedDeliveryModeState`, () => {
+    it(`should return the delivery modes`, (done) => {
+      spyOn(checkoutQuery, 'getCheckoutDetailsState').and.returnValue(
+        of(<QueryState<CheckoutState>>{
+          loading: false,
+          error: false,
+          data: {
+            deliveryMode: mockDeliveryMode,
+          },
+        })
+      );
+
+      service
+        .getSelectedDeliveryModeState()
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual(<QueryState<DeliveryMode | undefined>>{
+            loading: false,
+            error: false,
+            data: mockDeliveryMode,
+          });
+          done();
+        });
+    });
+  });
+
+  describe(`setDeliveryMode`, () => {
```

Comment:
should we add a case that says it(should throw error and NOT set/dispatch delivery mode, () => { when no deliveryModeCode is provided for the branches coverage ;d

Reply:
I actually decided to remove the check, as it shouldn't be allowed to pass a falsy value ;)